pkg/querier/batch/batch.go

```diff
@@ -156,16 +156,16 @@ func (a *iteratorAdapter) At() (int64, float64) {
 	return a.curr.Timestamps[a.curr.Index], a.curr.Values[a.curr.Index]
 }
 
-// AtHistogram implements chunkenc.Iterator.
-func (a *iteratorAdapter) AtHistogram() (int64, *histogram.Histogram) {
+// AtHistogram implements chunkenc.Iterator. It does not copy the underlying histogram as that optimization is left to the caller / higher level.
+func (a *iteratorAdapter) AtHistogram(*histogram.Histogram) (int64, *histogram.Histogram) {
 	if a.curr.ValueType != chunkenc.ValHistogram {
 		panic(fmt.Sprintf("Cannot read histogram from batch %v", a.curr.ValueType))
 	}
 	return a.curr.Timestamps[a.curr.Index], (*histogram.Histogram)(a.curr.PointerValues[a.curr.Index])
 }
 
-// AtFloatHistogram implements chunkenc.Iterator.
-func (a *iteratorAdapter) AtFloatHistogram() (int64, *histogram.FloatHistogram) {
+// AtFloatHistogram implements chunkenc.Iterator. It does not copy the underlying histogram as that optimization is left to the caller / higher level.
+func (a *iteratorAdapter) AtFloatHistogram(*histogram.FloatHistogram) (int64, *histogram.FloatHistogram) {
```

Comment:
What is the purpose of this `*histogram.Histogram` argument? It is not used in the function body.